cmake/public/xpu.cmake

```diff
@@ -0,0 +1,35 @@
+# ---[ xpu
+
+# Poor man's include guard
+if(TARGET torch::xpurt)
+  return()
+endif()
+
+# Find Intel SYCL Library.
+find_package(IntelSYCL REQUIRED)
+if(NOT IntelSYCL_FOUND)
+  set(CAFFE2_USE_XPU OFF)
+  return()
+endif()
+
+# Intel SYCL library interface
+add_library(XPU::sycl INTERFACE IMPORTED)
```

Comment:
It's SYCL runtime interface, not XPU backend specific. For example, if another backend bases on SYCL runtime, the backend can use it as well. So can we have another name for it? Like `torch::sycl`?

Reply:
torch::sycl